draft-ietf-jsonpath-base.md

```diff
@@ -314,7 +315,12 @@ $.store.book[0].title
 to build paths that are input to a JSONPath implementation.
 A single path may use a combination of bracket and dot notations.
 
-Dot notation is merely a shorthand way of writing certain bracket notations.
+Dot notation is a shorthand way of writing certain bracket notations.
+
+Note that in this specification, and for JSONPath expressions in general, when selecting
+object members by name, bracket notation is strictly preferred to dot notation. This is
```

Comment:
I don't agree with that statement.
(I do agree that $.foo.bar is two segments, not one.)

Reply:
I don't think the spec should prefer bracket notation to dot notation, even though the former is more general than the latter.

Reply:
Fair enough, but for **developers**, they should never create a JSONPath by saying
```
path = "$" + join(steps, ".") 
```

They should always use bracket notation.  So I think the spec would benefit from a caution to developers.

And if possible, I'd like to keep the foo/bar example because I've found it works very well in explaining this point  to people.

Reply:
I certainly like that example - see ef645cbe216243f28f861363c082db0ae1fdd57a.

Reply:
Please note I have not yet adopted the above caution to developers.